pkg/certificateprovider/azurekeyvault/provider.go

```diff
@@ -39,7 +41,7 @@ type keyvaultObject struct {
 }
 
 // returns an array of certificates based on certificate properties defined in attrib map
-func GetCertificates(ctx context.Context, attrib map[string]string) ([]types.Certificate, error) {
+func GetCertificates(ctx context.Context, attrib map[string]string) ([]*x509.Certificate, error) {
```

Comment:
One of the future items is to enable Certificate providers in the cosign verifier path. It looks like cosign load a [ECDSA](https://github.com/deislabs/ratify/blob/main/plugins/verifier/cosign/cosign.go#L150) key, will this interface work with cosign?